src/Core/Repositories/EntityFramework/CollectionRepository.cs

```diff
@@ -0,0 +1,237 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Bit.Core.Models.Data;
+using Bit.Core.Models.Table;
+using Bit.Core.Repositories.EntityFramework.Queries;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using EfModel = Bit.Core.Models.EntityFramework;
+using TableModel = Bit.Core.Models.Table;
+
+namespace Bit.Core.Repositories.EntityFramework
+{
+    public class CollectionRepository : Repository<TableModel.Collection, EfModel.Collection, Guid>, ICollectionRepository
+    {
+        public CollectionRepository(IServiceScopeFactory serviceScopeFactory, IMapper mapper)
+            : base(serviceScopeFactory, mapper, (DatabaseContext context) => context.Collections)
+        { }
+
+        public override async Task<TableModel.Collection> CreateAsync(Collection obj)
+        {
+            await base.CreateAsync(obj);
+            using (var scope = ServiceScopeFactory.CreateScope())
+            {
+                var dbContext = GetDatabaseContext(scope);
+                // TODO: User_BumpAccountRevisionDateByCollectionId
+            }
+            return obj;
+        }
+
+        public async Task CreateAsync(Collection obj, IEnumerable<SelectionReadOnly> groups)
+        {
+            await base.CreateAsync(obj);
+            using (var scope = ServiceScopeFactory.CreateScope())
+            {
+                var dbContext = GetDatabaseContext(scope);
+                var availibleGroups = await (from g in dbContext.Groups
+                                      where g.OrganizationId == obj.OrganizationId
+                                      select g.Id).ToListAsync();
+                var collectionGroups = groups
+                    .Where(g => availibleGroups.Contains(g.Id))
+                    .Select(g => new EfModel.CollectionGroup(){
+                        CollectionId = obj.Id,
+                        GroupId = g.Id,
+                        ReadOnly = g.ReadOnly,
+                        HidePasswords = g.HidePasswords
+                    });
+                Console.WriteLine(System.Text.Json.JsonSerializer.Serialize(collectionGroups, new System.Text.Json.JsonSerializerOptions(){WriteIndented = true}));
```

Comment:
This wasn't using a using because I was scoping out if System.Text.Json would meet all our needs without also using something like Newtonsoft. This line has been removed and the usings are cleaned up now.